R/assertions.R

```diff
@@ -1301,6 +1296,100 @@ assert_varval_list <- function(arg, # nolint
   invisible(arg)
 }
 
+#' Is an Argument a List of Expressions?
+#'
+#' Checks if the argument is a list of expressions.
+#'
+#' @param arg A function argument to be checked
+#' @param required_elements A `character` vector of names that must be present in `arg`
+#' @param named If set to `TRUE`, an error is issued if not all elements of the
+#'   list are named.
+#' @param optional Is the checked argument optional? If set to `FALSE` and `arg`
+#' is `NULL` then an error is thrown.
+#'
+#' @return
+#' The function throws an error if `arg` is not a list of expressions.
+#' Otherwise, the input it returned invisibly.
+#'
+#' @keywords assertion
+#' @family assertion
+#' @export
+#'
+#' @examples
+#' library(rlang)
+#'
+#' example_fun <- function(vars) {
+#'   assert_expr_list(vars)
+#' }
+#' example_fun(exprs(DTHDOM = "AE", DTHSEQ = AESEQ))
+#'
+#' try(example_fun(exprs("AE", DTSEQ = AESEQ, !!mean)))
+assert_expr_list <- function(arg, # nolint
+                             required_elements = NULL,
+                             named = FALSE,
+                             optional = FALSE) {
+  assert_logical_scalar(named)
+  assert_logical_scalar(optional)
+  assert_character_vector(required_elements, optional = TRUE)
+
+  if (optional && is.null(arg)) {
+    return(invisible(arg))
+  }
+
+  if (!inherits(arg, "list")) {
+    err_msg <- sprintf(
+      paste0(
+        "`%s` must be a named list of expressions but it is %s\n",
+        "\u2139 To create a list of expressions use `exprs()`"
+      ),
+      arg_name(substitute(arg)),
+      what_is_it(arg)
+    )
+    abort(err_msg)
+  }
+
+  if (named) {
+    assert_named(arg)
+  }
+
+  if (!is.null(required_elements)) {
+    missing_elements <- setdiff(required_elements, names(arg))
+    if (length(missing_elements) >= 1L) {
+      err_msg <- sprintf(
+        "The following required elements are missing in `%s`: %s",
+        arg_name(substitute(arg)),
+        enumerate(missing_elements, quote_fun = squote)
+      )
+      abort(err_msg)
+    }
+  }
+
+  invalids <- arg[!map_lgl(
+    arg,
+    ~ is_call(.x) || is_expression(.x)
+  )]
+
+  if (length(invalids) > 0) {
+    abort(
+      paste0(
+        "The elements of the list ",
+        arg_name(substitute(arg)),
+        " must be expressions.\n",
```

Comment:
```suggestion
      paste0(
        "The elements of the list ",
        arg_name(substitute(arg)),
        " must be an expression.\n",
```

Ugh...as a native English speaker I spent way too much time looking at this phrase.  @zdz2101 or @sadchla-codes is your first language English?? If so, does this one make sense?

![image](https://user-images.githubusercontent.com/10111024/231266699-78b3e1e5-2ac0-493b-a9e9-9b6e31d12652.png)